### ISSUE TITLE
Add DPT-hybrid

### ISSUE THREAD

**Issue by NielsRogge** [huggingface/transformers#20435]
### Model description

DPT-hybrid is used in Stable Diffusion 2.0's depth model and is also a nice depth estimation model in general.

We already support DPT, but not DPT-hybrid. The latter uses ViT-hybrid as backbone.

Hence, we first would need to add ViT-hybrid to the library. This model is very similar to a regular ViT, except that instead of patchifying an image and embedding each patch, this model uses a pre-trained ResNetv2 to embed the image before feeding the features to a Transformer encoder. 

This means that we first need to add ResNetv2 to the library. 😅 however that seems fairly doable, I'd recommend porting the one from timm: https://github.com/rwightman/pytorch-image-models/blob/main/timm/models/resnetv2.py.

### Open source status

- [X] The model implementation is available
- [X] The model weights are available

### Provide useful links for the implementation

DPT-hybrid is available here: https://github.com/isl-org/MiDaS

**Comment by nandwalritik** [comment]
Hi @NielsRogge Can I take this up if no one is working on it?

**Comment by patrickvonplaten** [comment]
This would be great @nandwalritik ! The Stable Diffusion Depth estimation model depends on it, so we'd definitely help you in whatever way we can :-)

**Comment by patrickvonplaten** [comment]
@NielsRogge should we port the ResNetv2 exactly like ResNetv1: https://github.com/huggingface/transformers/blob/main/src/transformers/models/resnet/modeling_resnet.py or could we directly port it from `timm`?

**Comment by patrickvonplaten** [comment]
Could you provide some more links on how to add ViT-Hybrid?

**Comment by NielsRogge** [comment]
@patrickvonplaten yes we can add ResNetv2 in the same way as we added Resnet V1. Meaning, a separate standalone model in the library.

Once we have that, we have everything we need to define modeling_dpt_hybrid.py. This would be a copy of modeling_dpt.py, except that we leverage ResNetv2 (we can do that using the new AutoBackbone API which is just added) in the DPTViTEmbeddings class. 
For stable diffusion there's no need to add hybrid ViT as a standalone model now (i.e. we can add modeling_hybrid_vit.py another time), it can be defined in modeling_dpt_hybrid.py. This file could copy everything from modeling_dpt.py, except that it includes the tweak in the embeddings layer (

**Comment by nandwalritik** [comment]
@NielsRogge @patrickvonplaten Can you let me know the next steps, what should I add first?
ResNetv2 -> ViT-hybrid -> DPT-hybrid, will this be the order?

**Comment by NielsRogge** [comment]
@nandwalritik yes, let's start by adding ResNetv2, based on timm's implementation found here: https://github.com/rwightman/pytorch-image-models/blob/main/timm/models/resnetv2.py.

Next, we can add DPT-hybrid, based on the modeling files here: https://github.com/isl-org/DPT/tree/main/dpt

**Comment by nandwalritik** [comment]
Ok I Will start with ResNetv2 and then I will try adding DPT-hybrid.

**Comment by nandwalritik** [comment]
Should I use `add-new-model` or `add-new-model-like` command as we already have ResNet implementation in huggingface ?

**Comment by NielsRogge** [comment]
You can use `add-new-model-like` and start from `resnet` (we might deprecate `add-new-model`)
